heclib/heclib_c/src/Internal/ztsRegStoreBlock.c

```diff
@@ -834,18 +834,51 @@ int ztsRegStoreBlock(long long *ifltab, zStructTimeSeries *tss, const char *path
 		internalHeader[INT_HEAD_blockStartPosition] = blockStartPosition;
 		internalHeader[INT_HEAD_blockEndPosition] = blockStartPosition + numberToStore - 1;
 		ipos = firstValid * valueSize;
-		status = ztsWriteBlock(ifltab, tss, pathname, 											
-								&ival, 0, numberToStore,  
-								&values[ipos],  valueSize,
-								quality,  qualityElementSize,
-								notes,  inoteElementSize,
-								cnotesToStore, cnotesToStoreLen,
-								profileDepths, profileDepthsSize,
-								internalHeader,  
-								userHeader,  userHeaderNumber,
-								0, numberInBlock,					
-								dataType);
 
+
+		zStructRecordBasics* rb = zstructRecordBasicsNew(pathname);
+		status = zgetRecordBasics(ifltab, rb);
+
+		int recordType = rb->recordType;
+		zstructFree(rb);
+
+		if (status == STATUS_RECORD_FOUND && ((recordType == DATA_TYPE_RTD && dataType == DATA_TYPE_RTS))) {
+
+			char start_date[12];
+			julianToDate(startJulian, 4, start_date, sizeof(start_date));
+			char start_time[10];
+			secondsToTimeString(startSeconds, 0, 2, start_time, sizeof(start_time));
+
+			double* values_double = calloc(numberToStore, sizeof(double));
+			if (values_double) {
+				convertDataArray((void*)&values[ipos], (void*)values_double, numberToStore, 1, 2);
+
+				zStructTimeSeries* tss_double = zstructTsNewRegDoubles(pathname, values_double, numberToStore, start_date, start_time, tss->units, tss->type);
+				tss_double->allocated[zSTRUCT_TS_doubleValues] = 1;
```

Comment:
Are we throwing away any quality codes, notes, supplemental info, etc... that were passed in?
I just now thought about this.